### PR TITLE
Add Bun to tests, and release Bun version of Binary and Docker image

### PR DESCRIPTION
**Pull request by ardatan**

### CODE REVIEW

File: .yarn/patches/graphql+16.9.0.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+       if (this._typeMap[typeName] !== undefined) {
+-        throw new Error(
+-          `Schema must contain uniquely named types but contains multiple types named "${typeName}".`,
+-        );
++        // throw new Error(
++        //   `Schema must contain uniquely named types but contains multiple types named "${typeName}".`,
++        // );
+       }
+ 
+       this._typeMap[typeName] = namedType;
```
Comment (enisdenjo) [inline .yarn/patches/graphql+16.9.0.patch:29]: We're doing something wrong if this patch is necessary. E2E's shouldn't be monkey patched like this.

File: e2e/tsconfig-paths/tsconfig-paths.e2e.ts [hunk 2]
```diff
@@ -2,9 +2,13 @@ import { createTenv, getAvailablePort } from '@internal/e2e';
...
 
-const { gateway, fs } = createTenv(__dirname);
+const { gateway, fs, gatewayRunner } = createTenv(__dirname);
 
-it('should start gateway', async () => {
+const skipIfIt = function (condition: boolean) {
+  return condition ? it.skip : it;
+};
+
+skipIfIt(gatewayRunner.includes('bun'))('should start gateway', async () => {
```
Comment (enisdenjo) [inline e2e/tsconfig-paths/tsconfig-paths.e2e.ts:11]: This is not necessary, [vitest supports `it.skipIf`](https://vitest.dev/api/#test-skipif).

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -37,35 +37,65 @@ jobs:
...
+            os: ubuntu-latest
+            gateway-runner: bun-bin
             node-version: 22
-    name: E2E / ${{matrix.setup.gateway-runner == 'node' && format('Node v{0}', matrix.setup.node-version) || ''}}${{matrix.setup.gateway-runner == 'docker' && 'Docker' || ''}}${{matrix.setup.gateway-runner == 'bin' && format('Binary on {0}', matrix.setup.os) || ''}}
+          - workflow-name: Bun Docker on Ubuntu
+            os: ubuntu-latest
+            gateway-runner: bun-docker
+            node-version: 22
+
+    name: E2E / ${{matrix.setup.workflow-name}}
```
Comment (enisdenjo) [inline .github/workflows/test.yml:94]: I love the `workflow-name`! Good thinking.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -54,13 +57,13 @@ jobs:
...
             const publishedPackages = ${{ needs.stable.outputs.publishedPackages || needs.snapshot.outputs.publishedPackages }};
-            const gateway = publishedPackages.find((p) => p.name === '@graphql-hive/gateway');
+            const gateway = publishedPackages.find((p) => p.name == '@graphql-hive/gateway');
             if (!gateway) {
               return core.setFailed('@graphql-hive/gateway was not published!');
             }
             const { version } = gateway;
             let r;
-            if (context.eventName === 'pull_request') {
+            if (context.eventName == 'pull_request') {
```
Comment (enisdenjo) [inline .github/workflows/release.yml:66]: Why?

File: .github/workflows/bench.yml [hunk 5]
```diff
@@ -36,5 +36,9 @@ jobs:
         continue-on-error: true
         with:
           key: docker-images-${{ runner.os }}-${{ steps.hash-docker-images.outputs.result }}
+      - name: Build
+        run: yarn build
+      - name: Relink
+        run: yarn install
```
Comment (enisdenjo) [inline .github/workflows/bench.yml:42]: Why was this necessary?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -54,13 +57,13 @@ jobs:
         with:
           script: |
             const publishedPackages = ${{ needs.stable.outputs.publishedPackages || needs.snapshot.outputs.publishedPackages }};
-            const gateway = publishedPackages.find((p) => p.name === '@graphql-hive/gateway');
+            const gateway = publishedPackages.find((p) => p.name == '@graphql-hive/gateway');
```
Comment (enisdenjo) [inline .github/workflows/release.yml:60]: Why?

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -107,16 +110,16 @@ jobs:
...
-          header: snapshot-release-docker-image
+          header: snapshot-release-${{ matrix.runtime == 'Bun' && 'bun-' || '' }}docker-image
           message: |
-            ### 🚀 Snapshot Release (Docker Image)
+            ### 🚀 Snapshot Release (${{matrix.runtime}} Docker Image)
 
             The latest changes of this PR are available as image on GitHub Container Registry (based on the declared `changesets`):
             ```
-            ghcr.io/graphql-hive/gateway:${{ fromJSON(steps.ver-gateway.outputs.result).version }}
+            ghcr.io/graphql-hive/gateway:${{ fromJSON(steps.ver-gateway.outputs.result).version }}${{ matrix.runtime == 'Bun' && '.bun' || '' }}
```
Comment (enisdenjo) [inline .github/workflows/release.yml:119]: Convention is to use a dash for different runtimes/OS. See [node dockerimages](https://hub.docker.com/_/node) (node-alpine, node-bookworm, etc.)

The reason dash is preferred is to avoid confusion with versions. For example, `gateway:1.2.bun` makes "bun" a patch version.

**Comment by enisdenjo** [comment]
Reverted with https://github.com/graphql-hive/gateway/commit/dc7f728d73062c6ac80d216594b98b8f89f44b42. Will be reopened in a new PR.

File: .github/workflows/release.yml (hunk 7 above)
Comment (ardatan) [reply .github/workflows/release.yml:119]: Done!

File: .github/workflows/release.yml (hunk 6 above)
Comment (ardatan) [reply .github/workflows/release.yml:60]: Bad find & replace

File: .github/workflows/bench.yml (hunk 5 above)
Comment (ardatan) [reply .github/workflows/bench.yml:42]: Bench workflows started failing with import errors to hive gateway :/

File: .github/workflows/release.yml (hunk 4 above)
Comment (ardatan) [reply .github/workflows/release.yml:66]: Bad find & replace

File: e2e/tsconfig-paths/tsconfig-paths.e2e.ts (hunk 2 above)
Comment (ardatan) [reply e2e/tsconfig-paths/tsconfig-paths.e2e.ts:11]: Done!
